tools/test.py

```diff
@@ -126,6 +126,12 @@
         all_tests = {}
         tests = {}
 
+        # Add the test macro, when compiling all test
+        if options.macros is None:
+           options.macros = ['MBED_TEST']
+        else:
+            options.macros.append(['MBED_TEST'])
```

Comment:
@deepikabhavnani this isn't right and will probably cause problems:

```
>>> a = [1, 2, 3]
>>> a.append([4])
>>> a
[1, 2, 3, [4]]
```

(Note the extra list)

It should be `options.macros.append('MBED_TEST')`